remote-workstations/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -1,122 +1,117 @@
 #!/bin/bash
 
-install_driver()
+if [ $# -eq 1 ]
+then
+    AGENT_TYPE="$1"
+else
+    # the first argument is the Registration Code of PCoIP agent
+    REGISTRATION_CODE="$1"
+    # the second argument is the agent type
+    AGENT_TYPE="$2"
+    # the third argument is the VM name
+    VM_NAME="$3"
+    # the forth argument is the domain name
+    DOMAIN_NAME="$4"
+    # the fifith argument is the username
+    USERNAME="$5"
+    # the sixth argument is the password
+    PASSWORD="$6"
+    # the seventh argument is the domain group to join
+    GROUP="$7"
+    # the eighth argument is the agent installer channel
+    AGENT_CHANNEL="$8"
+    # the ninth argument is the userStorageAccountUri
+    STORAGEURI="$9"
+    # the tenth argument is the sumo collector ID
+    COLLECTOR_ID="${10}"
+    # the eleventh argument is the SaS Storage account token
+    SAS_TOKEN="${11}"
+    # the twelfth argument is the Computer OU string
+    OU="${12}"
+fi
+
+update_kernel_dkms() 
 {
-    # Install kernel-source/kernel-devel and gcc
-    sudo yum -y install kernel-source kernel-devel gcc
-    FILE_NAME="NVIDIA-Linux-x86_64-367.106-grid.run"
-    FILE_LOCATION="/root/$FILE_NAME"
-    # Download Driver first
-    echo "Downloading and Installing Nvidia driver"
-    sudo wget --retry-connrefused --tries=3 --waitretry=5 -O $FILE_LOCATION https://binarystore.blob.core.windows.net/thirdparty/nvidia/$FILE_NAME
-    exitCode=$?
-    if [ $exitCode -ne 0 ]
+  sudo yum -y update
+
+  sudo yum -y install kernel-devel
+
+  sudo rpm -Uvh --quiet https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+
+  sudo yum -y install dkms
+}
+
+# need to reboot after install
+disable_nouveal()
+{
+    echo 'blacklist nouveau' | sudo tee -a /etc/modprobe.d/nouveau.conf
+    
+    echo 'blacklist lbm-nouveau' | sudo tee -a /etc/modprobe.d/nouveau.conf
+}
+
+# need to reboot after install Linux Integration Services for Hyper-V
+install_lis()
+{    
+    local LIS_FILE="lis-rpms-4.2.3-4.tar.gz"
+    
+    wget --retry-connrefused --tries=3 --waitretry=5  http://download.microsoft.com/download/6/8/F/68FE11B8-FAA4-4F8D-8C7D-74DA7F2CFC8C/$LIS_FILE
```

Comment:
I see a couple of things here...

* The URL should be enclosed in double quotes as it's a string
* The URL is using HTTP -- we should use HTTPS

Reply:
good catch. it is from original file